### DISCUSSION TITLE
SPDX SBOM generation

### DISCUSSION THREAD

**Discussion by stevemagness** [renovatebot/renovate#11780] [Suggest an Idea]
With the announcement that SPDX is now an ISO standard (https://spdx.dev/spdx-specification-is-now-an-iso-standard/), would the remit of renovatebot extend to generating an SPDX SBOM? Feels like it might be 'half way there' already because it knows about the dependencies?

**Comment by rarkins** [comment]
I think "half way there" is probably accurate. Renovate currently doesn't care enough about transitive dependencies - because it only cares about things it can update. It may not fit well into the CLI approach either, but it's something I'd consider adding as an export from the app dashboard.

**Reply by Silvanoc** [reply]
The Renovate scanning capabilities are amazing even if only for direct dependencies. IMO only that functionality might be very useful to get direct dependencies that can be used as an input for other tools capable of creating dependency trees including transitive dependencies.

Is there a way to let Renovate generate a machine-friendly dependency list (for example, in a JSON document) without getting the MRs?

**Reply by Silvanoc** [reply]
BTW I'm not thinking on a CLI approach, but rather a pipeline approach. For example, a pipeline that can create SBOMs for release tags.

**Reply by HonkingGoose** [reply]
### Self-service SBOM

You can generate a NTIA-compliant SBOM from any GitHub-hosted repository where you have `read` access. Read the GitHub blog for more information, two relevant quotes to get you started:

> Today, we’re happy to announce a new Export SBOM function that allows anyone with read access to a GitHub cloud repository to generate an [NTIA](https://ntia.gov/page/software-bill-materials)-compliant SBOM with a single click. [^self-service-blog]

> To generate an SBOM, simply click the new **Export SBOM button** on a repository’s dependency graph.

### Third-party actions to generate SBOM

> For example, a pipeline that can create SBOMs for release tags.

The GitHub docs mention some third-party actions that can generate a SBOM for you [^comparison-of-actions]. GitHub recommends the `sbom-generator-action`. [^github-maintained-action]

[^self-service-blog]: [GitHub Blog, Introducing self-service SBOMs](https://github.blog/2023-03-28-introducing-self-service-sboms/)
[^comparison-of-actions]: [GitHub Docs, Generating and submitting a SBOM](https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/using-the-dependency-submission-api#generating-and-submitting-a-software-bill-of-materials-sbom)
[^github-maintained-action]: [GitHub Marketplace, SBOM-generator-action](https://github.com/marketplace/actions/sbom-generator-action)

**Comment by setchy** [comment]
+1 - would really love to be able to generate SPDX SBOMs from my `renovate` pipeline (currently using the CLI + Bitbucket Pipelines)

**Comment by jamietanna** [comment]
FYI I've built[ a tool](https://dmd.tanna.dev/commands/renovate-to-sbom/) for this ([more info](https://www.jvt.me/posts/2023/11/03/renovate-to-sbom/)) that takes ie Renovate's debug logs and will then produce an SBOM - as above it'll only depend on the data that Renovate exposes, but can be a good start
